pkg-manager/resolve-dependencies/src/resolveDependencies.ts

```diff
@@ -1083,6 +1083,9 @@ async function resolveDependency (
     }
   }
   try {
+    if (!options.update && currentPkg.version) {
+      wantedDependency.pref = currentPkg.version
+    }
```

Comment:
This will only work with packages from the npm registry (not from github etc). Maybe if we would save the pref as the package id in the lockfile, we could solve it for all.